src/components/swap/GasBreakdownTooltip.tsx

```diff
@@ -62,6 +64,10 @@ export function GasBreakdownTooltip({
   const approvalEstimate = trade.approveInfo.needsApprove ? trade.approveInfo.approveGasEstimateUSD : undefined
   const wrapEstimate =
     isUniswapXTrade(trade) && trade.wrapInfo.needsWrap ? trade.wrapInfo.wrapGasEstimateUSD : undefined
+
+  const { chainId } = useWeb3React()
```

Comment:
the Swap component's chainId can technically be different than the user's connected chainId

use `trade.inputAmount.currency.chainId` instead